yarn-project/sequencer-client/src/block_builder/solo_block_builder.ts

```diff
@@ -416,9 +418,14 @@ export class SoloBlockBuilder implements BlockBuilder {
     );
   }
 
-  protected getKernelDataFor(tx: ProcessedTx) {
-    return new PublicKernelData(
-      tx.data,
+  protected getKernelDataFor(tx: ProcessedTx): RollupKernelData {
+    const inputs = new RollupKernelCircuitPublicInputs(
```

Comment:
Agreed, though I think if we had proving this would fail anyway since we presently overwrite kernel output with squashed writes. So this is no worse that what we presently do.